mod.ts

```diff
@@ -98,7 +102,8 @@ function getSignatureKey(options: Required<GetSignedUrlOptions>): string {
 
 function getUrl(options: Required<GetSignedUrlOptions>, queryParameters: URLSearchParams, signature: string): string {
   queryParameters.set('X-Amz-Signature', signature)
-  return `https://${options.bucketName}.${options.endpoint}${options.objectPath}?${new URLSearchParams(queryParameters).toString()}`
+
+  return `https://${options.endpoint}${options.path}?${new URLSearchParams(queryParameters).toString()}`
```

Comment:
Happy to switch it to this way around.

Reply:
Having used S3 more over the past months I have found an issue with this. Check out [this blog post](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/).

The current accepted style is the one you have used.

However, there are S3 solutions that do not support the new style. Also, running such a solution on premise requires the certificates for the instance to accept all subdomains (e.g. `*.minio.local`). This might be an issue. So running a [Minio](https://min.io/) instance on-premise would be way easier with the older _S3 path request style_.

So, to summarize, I believe that the new request style has to be the default. However, there should be a way to use the older path request style.

How would you suggest to implement it? For example, [`s3fs`](https://github.com/s3fs-fuse/s3fs-fuse) uses a parameter `use_path_request_style`.

Reply:
Thanks for the clarification! I like the `usePathRequestStyle` option you've added.